File: Dataset/Leetcode/valid/46/348.java
  public List<List<Integer>> XXX(int[] nums) {
        if (nums == null) return null;
        List<List<Integer>> list = new ArrayList<>();
        if (nums.length == 0) return list;
        dfs(0, nums, list);
        return list;
    }

    private void dfs(int idx, int[] nums, List<List<Integer>> list) {
        // 不能再往下搜索
        if (idx == nums.length) {
            List<Integer> result = new ArrayList<>();
            for (int value : nums) {
                result.add(value);
            }
            list.add(result);
            return;
        }

        // 枚举这一层所有可以做出的选择
        for (int i = idx; i < nums.length; i++) {
            swap(nums, idx, i);
            dfs(idx + 1, nums, list);
            swap(nums, idx, i);
        }
    }

    private void swap(int[] nums, int i, int j) {
        int tmp = nums[i];
        nums[i] = nums[j];
        nums[j] = tmp;
    }

File: Dataset/Leetcode/valid/48/348.cpp
class Solution {
public:
    void XXX(vector<vector<int>>& matrix) {
        int m=matrix.size(),n=matrix[0].size();
        for(int i=0;i<m/2;++i)
            for(int j=0;j<n;++j)
            {
                swap(matrix[i][j],matrix[m-i-1][j]);
            }
        for(int i=0;i<m;++i)
            for(int j=i+1;j<n;++j)
            {
                swap(matrix[i][j],matrix[j][i]);
            }
    }
};

